### PR TITLE
Reduce calls to getHeader in WebRequestImpl.determineIfRequestHasAuthenticationData

### PR DESCRIPTION
**Pull request by uberskigeek**
Performance improvement by reducing calls to getHeader in WebRequestImpl.determineIfRequestHasAuthenticationData

### CODE REVIEW

**Comment by uberskigeek** [comment]
#build

**Review by dettmoney — COMMENTED** [review]
The old code had null checks around hdrValue - is that not necessary anymore?

File: dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebRequestImpl.java [hunk 1]
```diff
@@ -150,13 +151,16 @@ public MatchResponse getMatchResponse() {
...
     }
 
-    private boolean isBasicAuthHeaderInRequest(HttpServletRequest request) {
-        String hdrValue = request.getHeader(BasicAuthAuthenticator.BASIC_AUTH_HEADER_NAME);
-        return hdrValue != null && hdrValue.startsWith("Basic ");
+    private boolean isBasicOrBearerAuthHeaderInRequest(String hdrValue) {
+        if ((hdrValue.startsWith(BASIC_AUTHORIZATION_METHOD)) || (hdrValue.startsWith(BEARER_AUTHORIZATION_METHOD))) {
+            return true;
+        } else
+            return false;
```
Comment (jhanders34) [inline dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebRequestImpl.java]:
```suggestion
    private boolean isBasicOrBearerAuthHeaderInRequest(String authHeaderValue) {
        return authHeaderValue != null && (authHeaderValue.startsWith(BASIC_AUTHORIZATION_METHOD) || authHeaderValue.startsWith(BEARER_AUTHORIZATION_METHOD));
```

**Review by jhanders34 — CHANGES_REQUESTED** [review]

**Review by jhanders34 — DISMISSED** [review]

**Comment by jhanders34** [comment]
#build

**Review by jhanders34 — DISMISSED** [review]

**Comment by uberskigeek** [comment]
#build

**Comment by uberskigeek** [comment]
#build

**Review by jhanders34 — DISMISSED** [review]

**Comment by uberskigeek** [comment]
#build

**Comment by uberskigeek** [comment]
#libby

**Review by jhanders34 — APPROVED** [review]

**Review by una-tapa — APPROVED** [review]
Refactoring looks good.  Thanks for letting us know.

**Pull request merged**